tests/test_endpoints.py

```diff
@@ -0,0 +1,232 @@
+    with vcr.use_cassette("test_databases_create.yaml") as cass:
+        response = cass._serializer.deserialize(cass.data[0][1]["content"])
+        return response["id"]
+
+
+def compare_cleaned_ids(id1: str, id2: str) -> bool:
+    return id1.replace("-", "").strip() == id2.replace("-", "").strip()
+
+
+@pytest.mark.vcr()
+def test_pages_create(client, page_id):
+    response = client.pages.create(
+        parent={"page_id": page_id},
+        properties={
+            "title": [{"text": {"content": TEST_PAGE_NAME}}],
+        },
+        children=[],
+    )
+
+    assert response["object"] == "page"
+    assert compare_cleaned_ids(response["parent"]["page_id"], page_id)
+
+
+@pytest.mark.vcr()
+def test_pages_retrieve(client, session_page_id, page_id):
+    response = client.pages.retrieve(page_id=session_page_id)
+    assert response["object"] == "page"
+    assert compare_cleaned_ids(response["parent"]["page_id"], page_id)
+
+
+@pytest.mark.vcr()
+def test_pages_update(client, session_page_id):
+    icon = {"type": "emoji", "emoji": "🛴"}
+
+    response = client.pages.update(page_id=session_page_id, icon=icon)
+    assert response["icon"] == icon
+
+
+@pytest.mark.vcr()
+def test_pages_properties_retrieve(client, session_page_id):
+    response = client.pages.properties.retrieve(
+        page_id=session_page_id, property_id="title"
+    )
+    assert response["results"][0]["type"] == "title"
+
+
+@pytest.mark.vcr()
+def test_blocks_children_create(client, session_page_id) -> str:
+    children = [
+        {"paragraph": {"rich_text": [{"text": {"content": "I'm a paragraph."}}]}}
+    ]
+
+    response = client.blocks.children.append(
+        block_id=session_page_id, children=children
+    )
+
+    assert response["object"] == "list"
+    assert len(response["results"]) > 0
+    assert response["results"][0]["id"]
+
+
+@pytest.mark.vcr()
+def test_blocks_children_list(client, session_page_id, session_block_id):
+    response = client.blocks.children.list(block_id=session_page_id)
+    assert response["object"] == "list"
+    assert response["type"] == "block"
+    assert compare_cleaned_ids(response["results"][0]["id"], session_block_id)
+
+
+@pytest.mark.vcr()
+def test_blocks_retrieve(client, session_block_id):
+    response = client.blocks.retrieve(block_id=session_block_id)
+    assert response["object"] == "block"
+    assert response["type"] == "paragraph"
+    assert compare_cleaned_ids(response["id"], session_block_id)
+
+
+@pytest.mark.vcr()
+def test_blocks_update(client, session_block_id):
+    new_plain_text = "I'm an updated paragraph."
+    new_text = {
+        "rich_text": [
+            {
+                "text": {"content": new_plain_text},
+                "annotations": {"bold": True, "color": "red_background"},
+            }
+        ]
+    }
+    response = client.blocks.update(block_id=session_block_id, paragraph=new_text)
+
+    assert response["paragraph"]["rich_text"][0]["plain_text"] == new_plain_text
```

Comment:
Yes, this is not going to work if we are using cassettes... I had to clean up the assertions because we can't compare the value
Again, feedbacks are welcomed about the best way to test these endpoints

Reply:
well, we probably want to make the tests autonomous, e.g. with a function-scoped fixture that creates a page just for that one specific test – AFAICT that's almost what you're doing already, but a function-scoped fixture would guarantee that no other test leak into this one